### PR TITLE
[KYUUBI #4112] [Authz] Check privilege for output table of InsertIntoHadoopFsRelationCommand

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

close issue #4122 .

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @yaooqinn @pan3793

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/gen/TableCommands.scala [hunk 1]
```diff
@@ -518,6 +518,19 @@ object TableCommands {
     TableCommandSpec(cmd, Nil, queryDescs = Seq(queryDesc))
   }
 
+  val InsertIntoHadoopFsRelationCommand = {
+    val cmd = "org.apache.spark.sql.execution.datasources.InsertIntoHadoopFsRelationCommand"
+    val actionTypeDesc = ActionTypeDesc(null, null, Some("UPDATE"))
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/gen/TableCommands.scala:523]: Shall we consider SaveMode？

Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/gen/TableCommands.scala:523]: OK, will add a SaveModeActionTypeExtractor to determine ActionType `INSERT` or `INSERT_OVERWRITE` in our case.

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/serde/actionTypeExtractors.scala [hunk 2]
```diff
@@ -40,3 +41,13 @@ class OverwriteOrInsertActionTypeExtractor extends ActionTypeExtractor {
     if (v1.asInstanceOf[Boolean]) INSERT_OVERWRITE else INSERT
   }
 }
+
+class SaveModeActionTypeExtractor extends ActionTypeExtractor {
+  override def apply(v1: AnyRef): PrivilegeObjectActionType = {
+    invokeAs[String](v1, "name") match {
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/serde/actionTypeExtractors.scala:50]: SaveMode is a stable API for end users, it's to match directly.

Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/serde/actionTypeExtractors.scala:50]: Do you mean import SaveMode directly and call the name method of it?

Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/serde/actionTypeExtractors.scala]: Addressed.

Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/serde/actionTypeExtractors.scala]: OK. Import `SaveMode` and use the name of `SaveMode` Enum for matching, while unable to match directly to Java enum in Scala.

**Review by yaooqinn — APPROVED** [review]
LGTM

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**